src/NuGet.Core/NuGet.Commands/SignCommand/CertificateProvider.cs

```diff
@@ -83,6 +86,9 @@ public static async Task<X509Certificate2Collection> GetCertificatesAsync(Certif
                                     options.CertificatePath)));
 
                         case CRYPT_E_NO_MATCH_HRESULT:
+#if NET9_0_OR_GREATER
+                        case CRYPT_E_BAD_DECODE:
+#endif
```

Comment:
@bartonjs @vcsjones FYI this was uncovered by the  [ExecuteCommandAsync_WithEmptyPkcs7File_RaisesErrorsOnceAsync](https://github.com/NuGet/NuGet.Client/blob/4a1faa2bcc771862a0ee1d241ebbfa1c196630ab/test/NuGet.Core.Tests/NuGet.Commands.Test/SignCommandRunnerTests.cs#L47) test, X509CertificateLoader returns a different hresult in this case.

Reply:
HResult values from exceptions aren't guaranteed (especially when they can change because of an OS change); so I don't anticipate putting it in the breaking change documentation.

Reply:
I'd agree with you in an ideal world, but this real world code makes me think we should at least give a heads up in the doc :)